trade-finance-manager-ui/server/controllers/helpers/deal-cancellation-enabled.helper.ts

```diff
@@ -19,11 +26,29 @@ export const canSubmissionTypeBeCancelled = (submissionType: DealSubmissionType)
  * @returns true or false depending on the feature flag, submission type and user type
  */
 export const isDealCancellationEnabled = (submissionType: DealSubmissionType, user: TfmSessionUser): boolean => {
-  // TODO: DTFS2-7298: also check that the deal hasn't already been cancelled.
-
   const isUserAllowedToCancelDeal = userIsInTeam(user, [TEAM_IDS.PIM]);
   const isDealCancellationFeatureFlagEnabled = isTfmDealCancellationFeatureFlagEnabled();
   const isAcceptableSubmissionType = canSubmissionTypeBeCancelled(submissionType);
 
   return isUserAllowedToCancelDeal && isDealCancellationFeatureFlagEnabled && isAcceptableSubmissionType;
 };
+
+/**
+ * Checks if a deal can be cancelled; if it has already been cancelled then this returns false
+ * @param cancellationStatus - optional current deal cancellation status
+ * @returns true or false depending on if the deal has already been cancelled
+ */
+export const canDealStillBeCancelled = (cancellationStatus?: TfmDealCancellationStatus): boolean => {
```

Comment:
What's the reason for `canDealStillBeCancelled` please, how about `canDealBeCancelled`?

Reply:
Fair point, have changed! Thank you